### ISSUE TITLE
Panic when constructing `footnote.entry` manually

### ISSUE THREAD

**Issue by urbanij** [typst/typst#2328]
### Error:
```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', crates/typst-library/src/meta/footnote.rs:271:37
```
### Reproduction code:

```
HI#footnote.entry(clearance: 2.5em)[There]
```

### git commit
`b584617c8d594cf2ba0ee5d3dd4a042eb65fc4a3`

### Operating system

macOS

### Typst version

- [X] I am using the latest version of Typst

**Comment by Dherse** [comment]
I am pretty sure you're not supposed to instantiate `footnote.entry` yourself, mind you it shouldn't crash but give an error when you do it yourself instead.

**Comment by Dherse** [comment]
Fixed in #2687
